src/Microsoft.OpenApi/Extensions/StringExtensions.cs

```diff
@@ -16,22 +17,21 @@ public static class StringExtensions
         /// Gets the enum value based on the given enum type and display name.
         /// </summary>
         /// <param name="displayName">The display name.</param>
-        public static T GetEnumFromDisplayName<T>(this string displayName)
+        [UnconditionalSuppressMessage("Trimming", "IL2075", Justification = "Fields are never trimmed for enum types.")]
+        public static T GetEnumFromDisplayName<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields)] T>(this string displayName)
         {
             var type = typeof(T);
             if (!type.IsEnum)
             {
                 return default;
             }
 
-            foreach (var value in Enum.GetValues(type))
+            foreach (var field in type.GetFields(BindingFlags.Public | BindingFlags.Static))
```

Comment:
Does this have any functional changes from the existing code? Specifically with a `Flags` enum?

Reply:
AFAICT, behavior seems to be the same between this implementation and the prior one for `[Flags]` enums. I'll add test coverage for this. FWIW, primary consumers of this API are non-Flags enums in Microsoft.OpenApi.Readers although since it is public API nothing prohibits end-users from using it. :/